.github/workflows/ci_tests.yaml

```diff
@@ -79,7 +81,11 @@ jobs:
 
       # Install GMT and other required dependencies from conda-forge
       - name: Install dependencies
-        run: conda env update --file environment.yml
+        run: |
+          conda install gmt=6.1.1 numpy pandas xarray netCDF4 packaging \
+                        make codecov coverage[toml] \
+                        pytest-cov pytest-mpl pytest>=6.0 \
+                        ipython
```

Comment:
How about splitting `environment.yml` into three files under `.github/workflows`:

- environment-base.yml (core packages, i.e. `gmt` ... `packaging`)
- environment-dev.yml (`make`, `codecov`, `pytest`, etc)
- environment-docs.yml (`sphinx`, `myst-parser`, etc)

That way we can do `conda install -f environment-base.yml -f environment-dev.yml` under 'ci_tests.yaml' and do `conda install -f environment-base.yml -f environment-docs.yml` under `docs.yaml` (which I would rename to `ci_docs.yaml` actually).

Probably best to keep the `environment.yml` with **all** the packages in the top-level still, so that new developers can do `conda env create` and get all the necessary dependencies installed locally. But for the Github Actions CI, it should be ok to split them up. Happy to discuss this a bit more :smile:

Reply:
It seems the CI running times don't change using `conda-lock` (#1055). Three separate environment.yml files seem to be the easiest way.

Reply:
> It seems the CI running times don't change using `conda-lock` (#1055). Three separate environment.yml files seem to be the easiest way.

That PR was actually using `poetry` (Alternative 2) and not `conda-lock` (Alternative 1), but yeah, point taken. Where do we want to put all of these environment*.yml files? I know you prefer to keep the top-level directory clean, so should we place them under `.github/workflows`, under a new `ci/` folder, or somewhere else?

Reply:
> Where do we want to put all of these environment*.yml files? I know you prefer to keep the top-level directory clean, so should we place them under `.github/workflows/`, under a new `ci/` folder, or somewhere else?

`ci_docs.yml` will use `envrionment_base.yml` and `environment_docs.yml`, and `ci_tests.yml` will use `envrionment_base.yml` and `environment_dev.yml`.

I actually feel better to just write the dependencies in the `ci_*.yml` workflows, instead of adding three new files to the repository.

Reply:
> I actually feel better to just write the dependencies in the `ci_*.yml` workflows, instead of adding three new files to the repository.

Sure, we could just put them in the ci_*.yml files. Sorry for the big detour!